app/queues/dns/dns-flow.server.ts

```diff
@@ -0,0 +1,88 @@
+import { FlowProducer } from 'bullmq';
+import {
+  addDbRecordQueueName,
+  createDnsRecordQueueName,
+  checkDnsStatusQueueName,
+  syncDbStatusQueueName,
+} from './create-record-worker.server';
+import { redis } from '~/lib/redis.server';
+
+import type { FlowJob } from 'bullmq';
+import type { DomainRecord } from '~/lib/domains.server';
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const addDnsRequest = async ({
+  username,
+  type,
+  name,
+  value,
+}: Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>) => {
+  // Step 1. Create a record in MySQL for a domain with pending status
+  const addDbRecord: FlowJob = {
+    name: `addDbRecord:${name}-${username}`,
+    queueName: addDbRecordQueueName,
+    data: { username, type, name, value } as Pick<
```

Comment:
I added the `as` part here, so the worker itself can dictate what data it needs. I suggest to export the required props from the worker file itself and use that here, I don't see this pick adding any value (correct me if I'm wrong)

Reply:
Please correct the other ones too in this file

Reply:
I'm not sure what is the best approach, but my thinking was that I have `...Record` types with similar props for DB, Route53 and Job. So I was thinking this is good to modify something that is available rather than thinking about reduced scope of props with ambiguous name. I should name my new type `JobRecord`.